metro-plugin-typescript throws false errors

### What happened?

metro-plugin-typescript throws unexpected false errors such as:
`error TS2305: Module '"react-native-screens"' has no exported member 'enableScreens'.`
tsc doesn't throws these errors nor does VS Code.

I'm using Expo 48:
```js
const { getDefaultConfig } = require("expo/metro-config");
const { TypeScriptPlugin } = require("@rnx-kit/metro-plugin-typescript");

module.exports = (() => {
  const config = getDefaultConfig(__dirname);
  config.serializer = {
    experimentalSerializerHook: TypeScriptPlugin(),
  };
  return config;
})();
```


### Affected Package

@rnx-kit/metro-plugin-typescript

### Version

0.3.3

### Which platforms are you seeing this issue on?

- [ ] Android
- [ ] iOS
- [ ] macOS
- [X] Windows

### System Information

```true
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i9-12900H
    Memory: 11.06 GB / 15.62 GB
    Shell: 3.3.1 - /usr/bin/fish
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
    Watchman: Not Found
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: Not Found
  Languages:
    Java: 11.0.17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0
    react-native: 0.71.3 => 0.71.3
  npmGlobalPackages:
    *react-native*: Not Found
```


### Steps to Reproduce

add react-native-screens to your project and add
`import { enableScreens } from "react-native-screens";`


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/.changeset/loud-avocados-know.md
+++ b/.changeset/loud-avocados-know.md
@@ -1,0 +1,5 @@
+---
+"@rnx-kit/metro-plugin-typescript": patch
+---
+
+Don't apply module suffixes when resolving main fields

--- a/packages/metro-plugin-typescript/src/resolver.ts
+++ b/packages/metro-plugin-typescript/src/resolver.ts
@@ -1,7 +1,12 @@
+import { parseModuleRef } from "@rnx-kit/tools-node";
 import intersection from "lodash/intersection";
 import isEqual from "lodash/isEqual";
 import ts from "typescript";
 import type { ResolverContext } from "./types";
+
+function isPackageRef(name: string): boolean {
+  return !parseModuleRef(name).path;
+}
 
 /**
  * Get TypeScript compiler options with the `moduleSuffixes` property configured
@@ -97,13 +102,17 @@ export function resolveModuleName(
   redirectedReference: ts.ResolvedProjectReference | undefined,
   options: ts.CompilerOptions
 ): ts.ResolvedModuleFull | undefined {
-  //  Ensure the compiler options has `moduleSuffixes` set correctly for this RN project.
-  const optionsWithSuffixes = getCompilerOptionsWithReactNativeModuleSuffixes(
-    context,
-    moduleName,
-    containingFile,
-    options
-  );
+  // Ensure the compiler options has `moduleSuffixes` set correctly for this RN
+  // project. If `moduleName` points to a package (no paths), don't add platform
+  // suffixes as they are not used when looking at main fields.
+  const optionsWithSuffixes = isPackageRef(moduleName)
+    ? options
+    : getCompilerOptionsWithReactNativeModuleSuffixes(
+        context,
+        moduleName,
+        containingFile,
+        options
+      );
 
   //
   //  Invoke the built-in TypeScript module resolver.

--- a/packages/metro-plugin-typescript/test/resolver.test.ts
+++ b/packages/metro-plugin-typescript/test/resolver.test.ts
@@ -12,7 +12,8 @@ describe("Resolver", () => {
     disableReactNativePackageSubstitution: false,
     platform: "ios",
     platformFileExtensions: [".ios", ".native", ""],
-    replaceReactNativePackageName: (x: string) => x + "-replaced",
+    replaceReactNativePackageName: (x: string) =>
+      x.replace(/^@rnx-kit/, "@replaced"),
   };
 
   test("resolves modules", () => {
@@ -20,15 +21,24 @@ describe("Resolver", () => {
     const mockTsResolveModuleName = jest.fn();
     ts.resolveModuleName = mockTsResolveModuleName;
     try {
-      mockTsResolveModuleName
-        .mockReturnValueOnce({ resolvedModule: { resolvedFileName: "first" } })
-        .mockReturnValueOnce({
-          resolvedModule: { resolvedFileName: "second" },
-        });
+      const moduleNames = [
+        "pkg-dir",
+        "pkg-up",
+        "pkg-up/index.js",
+        "@rnx-kit/tools-node",
+        "@rnx-kit/tools-node/lib/index.js",
+        "./index",
+      ];
+
+      const resolvedModules = moduleNames.map((name) => {
+        const resolvedModule = { resolvedFileName: name };
+        mockTsResolveModuleName.mockReturnValueOnce({ resolvedModule });
+        return resolvedModule;
+      });
 
       const modules = resolveModuleNames(
         context,
-        ["pkg-dir", "pkg-up"],
+        moduleNames,
         "/repos/rnx-kit/packages/test-app/src/app.ts",
         undefined,
         undefined,
@@ -36,25 +46,25 @@ describe("Resolver", () => {
         undefined
       );
 
-      expect(mockTsResolveModuleName).toHaveBeenCalledTimes(2);
+      expect(mockTsResolveModuleName).toHaveBeenCalledTimes(moduleNames.length);
       const calls = mockTsResolveModuleName.mock.calls;
       // 1st param: module name
-      expect(calls[0][0]).toEqual("pkg-dir-replaced");
-      expect(calls[1][0]).toEqual("pkg-up-replaced");
+      expect(calls[0][0]).toEqual("pkg-dir");
+      expect(calls[1][0]).toEqual("pkg-up");
+      expect(calls[2][0]).toEqual("pkg-up/index.js");
+      expect(calls[3][0]).toEqual("@replaced/tools-node");
+      expect(calls[4][0]).toEqual("@replaced/tools-node/lib/index.js");
+      expect(calls[5][0]).toEqual("./index");
       // 3rd param: CompilerOptions
-      expect(calls[0][2]).toEqual({
-        moduleSuffixes: [".ios", ".native", ""],
-      });
-      expect(calls[1][2]).toEqual({
-        moduleSuffixes: [".ios", ".native", ""],
-      });
+      expect(calls[0][2]).toEqual({});
+      expect(calls[1][2]).toEqual({});
+      expect(calls[2][2]).toEqual({ moduleSuffixes: [".ios", ".native", ""] });
+      expect(calls[3][2]).toEqual({});
+      expect(calls[4][2]).toEqual({ moduleSuffixes: [".ios", ".native", ""] });
+      expect(calls[5][2]).toEqual({ moduleSuffixes: [".ios", ".native", ""] });
 
       expect(Array.isArray(modules)).toBe(true);
-      expect(modules.length).toBe(2);
-      expect(modules).toEqual([
-        { resolvedFileName: "first" },
-        { resolvedFileName: "second" },
-      ]);
+      expect(modules).toEqual(resolvedModules);
     } finally {
       ts.resolveModuleName = origTsResolveModuleName;
     }

--- a/packages/test-app/src/App.native.tsx
+++ b/packages/test-app/src/App.native.tsx
@@ -1,8 +1,8 @@
 import { acquireTokenWithScopes } from "@rnx-kit/react-native-auth";
 import React, { useCallback, useMemo, useState } from "react";
 import {
-  Pressable,
   NativeModules,
+  Pressable,
   SafeAreaView,
   ScrollView,
   StatusBar,

--- a/packages/tools-node/src/module.ts
+++ b/packages/tools-node/src/module.ts
@@ -36,7 +36,7 @@ export function parseModuleRef(r: string): PackageModuleRef | FileModuleRef {
   const ref: PackageModuleRef = parsePackageRef(r);
 
   const indexPath = ref.name.indexOf("/");
-  if (indexPath > -1) {
+  if (indexPath >= 0) {
     const p = ref.name.substring(indexPath + 1);
     if (p) {
       ref.path = p;

